core/src/main/java/org/apache/calcite/rel/rules/materialize/MaterializedViewRule.java

```diff
@@ -1252,6 +1285,193 @@ protected RexNode replaceWithOriginalReferences(final RexBuilder rexBuilder,
+      RexNode truncatedLiteral = rexBuilder.makeCall(floorCall.getType(),
+          floorOperator, ImmutableList.of(literalOperand, floorCall.getOperands().get(1)));
+      Comparable v0 = RexInterpreter.evaluate(truncatedLiteral, Collections.emptyMap());
+      if (v0 == null) {
+        throw new AssertionError("interpreter returned null for " + v0);
+      }
+      Comparable v1 = RexInterpreter.evaluate(literalOperand, Collections.emptyMap());
+      if (v1 == null) {
+        throw new AssertionError("interpreter returned null for " + v1);
+      }
+      long modifiedVal = (long) v0;
+      final long originalVal = (long) v1;
+      // Since the view contains a FLOOR() if the query does a > or <= and the operand is already
+      // a FLOORED value we have to shift the value to the next higher or lower floored value
+      // If the query contains a >= or < we don't need to shift the modified value
+      if (shiftModifiedVal && modifiedVal == originalVal) {
+        final RexLiteral literal = (RexLiteral) floorCall.getOperands().get(1);
+        final TimeUnitRange unit = castNonNull(literal.getValueAs(TimeUnitRange.class));
+        if (isLowerBound) {
+          modifiedVal += unit.startUnit.multiplier.longValue();
+        } else {
+          modifiedVal -= unit.startUnit.multiplier.longValue();
+        }
+      }
+      return modifiedVal;
+    }
+
+    /**
+     * Generates a predicate that is added to the view so that we can use union rewriting.
+     *
+     * @param call the query predicate
+     * @return the predicate that is to be used as the view predicate or a view filter
+     */
+    @SuppressWarnings("BetaApi")
+    @Override public RexNode visitCall(RexCall call) {
+      switch (call.getKind()) {
+      case SEARCH:
+        Sarg sarg = castNonNull(((RexLiteral) call.getOperands().get(1)).getValueAs(Sarg.class));
+        if (sarg.rangeSet.asRanges().size() != 1) {
+          return super.visitCall(call);
+        }
+        Range r = (Range) sarg.rangeSet.asRanges().iterator().next();
+        final int floorIndex = ((RexInputRef) floorCall.getOperands().get(0)).getIndex();
+        RexNode tableInputRefOperand = call.operands.get(0);
+        if (!(tableInputRefOperand instanceof RexTableInputRef)) {
+          // ignore SARG that isn't on table column
+          return rexBuilder.makeLiteral(true);
+        }
+        int predicateIndex = ((RexTableInputRef) tableInputRefOperand).getIndex();
+        if (floorIndex != predicateIndex) {
+          // ignore predicates on different columns
+          return rexBuilder.makeLiteral(true);
+        }
+        if (!(r.lowerEndpoint() instanceof TimestampString)
+            || !(r.upperEndpoint() instanceof TimestampString)) {
+          // ignore predicates where the type isn't a timestamp
+          return rexBuilder.makeLiteral(true);
+        }
+        RexNode lb = rexBuilder.makeTimestampLiteral((TimestampString) r.lowerEndpoint(), 0);
```

Comment:
Please avoid cryptic names like `lb`, `ub`, there is already a lot of complex logic in these methods, let's try to help the code reader with more explicit names like `lowerBound` and `upperBound`. 

They are surely longer, but nowadays we have IDEs for that :)

Reply:
done